From 70cf1aacb024bccd8f54d3dc0ef4ea824bdd6faa Mon Sep 17 00:00:00 2001
From: Sachin Kamat <sachin.kamat@linaro.org>
Date: Mon, 19 Aug 2013 11:58:42 +0530
Subject: [PATCH 0987/1302] thermal: exynos: Fix potential NULL pointer
 dereference

NULL pointer was being dereferenced in its own error message.
Changed it to the correct device pointer.

Signed-off-by: Sachin Kamat <sachin.kamat@linaro.org>
Signed-off-by: Eduardo Valentin <eduardo.valentin@ti.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/thermal/samsung/exynos_thermal_common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/thermal/samsung/exynos_thermal_common.c b/drivers/thermal/samsung/exynos_thermal_common.c
index 4d8e444..f10a6ad 100644
--- a/drivers/thermal/samsung/exynos_thermal_common.c
+++ b/drivers/thermal/samsung/exynos_thermal_common.c
@@ -53,7 +53,7 @@ static int exynos_set_mode(struct thermal_zone_device *thermal,
 {
 	struct exynos_thermal_zone *th_zone = thermal->devdata;
 	if (!th_zone) {
-		dev_err(th_zone->sensor_conf->dev,
+		dev_err(&thermal->device,
 			"thermal zone not registered\n");
 		return 0;
 	}
@@ -231,7 +231,7 @@ static int exynos_get_temp(struct thermal_zone_device *thermal,
 	void *data;
 
 	if (!th_zone->sensor_conf) {
-		dev_err(th_zone->sensor_conf->dev,
+		dev_err(&thermal->device,
 			"Temperature sensor not initialised\n");
 		return -EINVAL;
 	}
@@ -251,7 +251,7 @@ static int exynos_set_emul_temp(struct thermal_zone_device *thermal,
 	struct exynos_thermal_zone *th_zone = thermal->devdata;
 
 	if (!th_zone->sensor_conf) {
-		dev_err(th_zone->sensor_conf->dev,
+		dev_err(&thermal->device,
 			"Temperature sensor not initialised\n");
 		return -EINVAL;
 	}
-- 
1.8.3.2

